packages/lib-utils/src/app/api-discovery/discovery-cache.ts

```diff
@@ -1,23 +1,51 @@
 import { consoleLogger } from '@monorepo/common';
+import * as _ from 'lodash-es';
+import { getReferenceForModel } from '../../k8s/k8s-utils';
```

Comment:
@karelhala this style of resource reference (tilda `~`) is deprecated. I'd like to avoid relying on it more if possible.

Reply:
We can avoid it, but we need to construct a key by which to merge the arrays, the models are defined by group, version and kind so I found this convinient to use. We could construct a key like `${group}/${version}-${kind}` to avoid using the deprecated thing.

Reply:
I think it's OK to use a composite group/version/kind string as a key, since each element is a string on its own.

@florkbr AFAIK, Console doesn't want to promote `group~version~kind` key as a public facing API but it's OK to use it internally, within Console app or within our SDK utilities.

Reply:
@karelhala @florkbr In future we can consider renaming `getReference` to `getInternalReference` and `getReferenceForModel` to `getInternalReferenceForModel` while undeprecating them and ensuring they are not exposed to SDK consumers.

Reply:
Agreed - since this is internal use we can keep as is